### PR TITLE
feat: Enforce id limits in instance emulator

### PR DESCRIPTION
**Pull request by amitchauhan1007**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5642)

### CODE REVIEW

File: google/cloud/bigtable/tests/instance_admin_emulator.cc [hunk 1]
```diff
@@ -42,6 +42,13 @@ class InstanceAdminEmulator final
     google::protobuf::TextFormat::PrintToString(*request, &request_text);
     std::cout << __func__ << "() request=" << request_text << "\n";
 
+    constexpr int kMaxInstanceIdLength = 33;
```
Comment (coryan) [inline google/cloud/bigtable/tests/instance_admin_emulator.cc:45]: The project is "east const" and we prefer `auto`: s/constexpr int/auto constexpr/ here and below.

**Review by coryan — COMMENTED** [review]
Looks good, one nit about the definition of the constants.

File: google/cloud/bigtable/tests/instance_admin_emulator.cc [hunk 2]
```diff
@@ -42,6 +42,13 @@ class InstanceAdminEmulator final
...
+    constexpr int kMinInstanceIdLength = 6;
+    if (((request->instance_id()).size() > kMaxInstanceIdLength) &&
```
Comment (devbww) [inline google/cloud/bigtable/tests/instance_admin_emulator.cc:47]: I presume these should all be disjunctions.

**Review by devbww — CHANGES_REQUESTED** [review]

File: google/cloud/bigtable/tests/instance_admin_emulator.cc [hunk 3]
```diff
@@ -42,6 +42,13 @@ class InstanceAdminEmulator final
...
+        ((request->instance_id()).size() < kMinInstanceIdLength)) {
+      throw std::invalid_argument(
```
Comment (devbww) [inline google/cloud/bigtable/tests/instance_admin_emulator.cc:49]: Raw `throw`s don't compile in the "noex" builds.  Consider `google/cloud/internal/throw_delegate.h`.

**Review by devbww — CHANGES_REQUESTED** [review]

**Comment by coryan** [comment]
@amitchauhan1007 no longer works with us, I will close this PR and send a new one with these changes.

**Pull request closed without merging**